src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -606,13 +606,54 @@ internal static bool TryGetDebuggerDisplayInfo(this DkmClrValue value, out Debug
             return displayInfo.HasValues;
         }
 
+        /// <summary>
+        /// Custom comparer for the GetDebuggerCustomUIVisualizerInfo method so that our custom set can
+        /// eliminate duplicated visualizer attributes from a type due to inheritance.
+        /// </summary>
+        private class DkmClrDebuggerVisualizerAttributeComparer : IEqualityComparer<DkmClrDebuggerVisualizerAttribute>
+        {
+            public bool Equals(DkmClrDebuggerVisualizerAttribute x, DkmClrDebuggerVisualizerAttribute y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                return x.VisualizerDescription == y.VisualizerDescription &&
+                       x.UISideVisualizerTypeName == y.UISideVisualizerTypeName &&
+                       x.UISideVisualizerAssemblyName == y.UISideVisualizerAssemblyName &&
+                       x.UISideVisualizerAssemblyLocation == y.UISideVisualizerAssemblyLocation &&
+                       x.DebuggeeSideVisualizerTypeName == y.DebuggeeSideVisualizerTypeName &&
+                       x.DebuggeeSideVisualizerAssemblyName == y.DebuggeeSideVisualizerAssemblyName;
+            }
+
+            public int GetHashCode(DkmClrDebuggerVisualizerAttribute obj)
+            {
+                const int prime = 31;
+
+                int hash = 1;
+                if (obj != null)
+                {
+                    hash = (hash * prime) + obj.VisualizerDescription.GetHashCode();
+                    hash = (hash * prime) + obj.UISideVisualizerTypeName.GetHashCode();
+                    hash = (hash * prime) + obj.UISideVisualizerAssemblyName.GetHashCode();
+                    hash = (hash * prime) + obj.UISideVisualizerAssemblyLocation.GetHashCode();
+                    hash = (hash * prime) + obj.DebuggeeSideVisualizerTypeName.GetHashCode();
+                    hash = (hash * prime) + obj.DebuggeeSideVisualizerAssemblyName.GetHashCode();
+                }
+
+                return hash;
+            }
+        }
+
         /// <summary>
         /// Returns the array of <see cref="DkmCustomUIVisualizerInfo"/> objects of the type from its <see cref="DkmClrDebuggerVisualizerAttribute"/> attributes,
         /// or null if the type has no [DebuggerVisualizer] attributes associated with it.
         /// </summary>
         internal static DkmCustomUIVisualizerInfo[] GetDebuggerCustomUIVisualizerInfo(this DkmClrType type)
         {
-            var builder = ArrayBuilder<DkmCustomUIVisualizerInfo>.GetInstance();
+            var visualizerAttributeBuilder =
+                HashSetBuilder<DkmClrDebuggerVisualizerAttribute, DkmClrDebuggerVisualizerAttributeComparer>.GetInstance();
```

Comment:
i woudl just make a pooled dictionary here.

Reply:
Sorry if this is obvious, but it appears that PooledDictionary is being built for .net standard 1.3 which corresponds to .NET Framework 4.6 and linking it brings a dependency on ImmutableDictionary that is .net standard 2.0 wouldn't this be redoing what I tried to do before? Or am I missing something?

Reply:
ResultProvider.csproj is now built with netstandard2.0 so `HashSet<T>` should be available.